### PR TITLE
Convert Class Component into Functional component

### PR DESCRIPTION
**Pull request by devhimanshuu**
ErrorBoundary functional components updated with Material Ui latest version

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by vishalvivekm** [comment]
Thank you @devhimanshuu 
Let's discuss this on Meshery Dev call, add this as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit) , if you would, please.

**Review by abhishekmmgn — COMMENTED** [review]
This is good. Thank you for the PR.

File: ui/components/ErrorBoundary.js [hunk 1]
```diff
@@ -1,40 +1,43 @@
...
 
-  resetErrorBoundary = () => {
-    this.setState({ hasError: false, error: null });
+    const resetErrorBoundary = () => {
+      setHasError(false);
+      setError(null);
+    };
+
+    window.addEventListener('error', handleErrors);
+    return () => window.removeEventListener('error', handleErrors);
```
Comment (Yashsharma1911) [inline ui/components/ErrorBoundary.js:20]: We don't need to listen to event, @devhimanshuu is there specific reason to use this, how you make sure we don't use error boundaries for all error which can be ignored

**Review by Yashsharma1911 — CHANGES_REQUESTED** [review]
Hey @devhimanshuu thanks for raising this PR, let's put your effort on right by by depricating this component, we already have a react boundary defined here, so can help in making sure that we delete this `ErrorBoundary` component and where ever we are using it, we replace with this [ErrorBoundaries](https://github.com/meshery/meshery/blob/387f71324f66e07b1bcb19ddb3bc9a6ca0cadda1/ui/components/General/ErrorBoundary.js#L33)

**Comment by Yashsharma1911** [comment]
Lint checks are failing

File: ui/components/ErrorBoundary.js (hunk 1 above)
Comment (devhimanshuu) [reply ui/components/ErrorBoundary.js:20]: I agree what you are saying but, i add these event listener for some global errors or unexpected error.  may be there is some optimized way of handling these errors.

**Comment by leecalcote** [comment]
Thank you, @devhimanshuu

**Pull request closed without merging**